satpy/writers/__init__.py

```diff
@@ -232,56 +236,46 @@ def add_overlay(orig, area, coast_dir, color=(0, 0, 0), width=0.5, resolution=No
         area = get_area_def(area)
     LOG.info("Add coastlines and political borders to image.")
 
-    if resolution is None:
-
-        x_resolution = ((area.area_extent[2] -
-                         area.area_extent[0]) /
-                        area.x_size)
-        y_resolution = ((area.area_extent[3] -
-                         area.area_extent[1]) /
-                        area.y_size)
-        res = min(x_resolution, y_resolution)
-
-        if res > 25000:
-            resolution = "c"
-        elif res > 5000:
-            resolution = "l"
-        elif res > 1000:
-            resolution = "i"
-        elif res > 200:
-            resolution = "h"
-        else:
-            resolution = "f"
-
-        LOG.debug("Automagically choose resolution %s", resolution)
+    old_args = [color, width, resolution, grid, level_coast, level_borders]
+    if any(arg is None for arg in old_args):
+        warnings.warn("'color', 'width', 'resolution', 'grid', 'level_coast', 'level_borders'"
+                      " arguments will be deprecated soon. Please use 'overlays' instead.", DeprecationWarning)
 
-    if hasattr(orig, 'convert'):
+    if hasattr(orig_img, 'convert'):
         # image must be in RGB space to work with pycoast/pydecorate
-        orig = orig.convert('RGBA' if orig.mode.endswith('A') else 'RGB')
+        res_mode = ('RGBA' if orig_img.mode.endswith('A') else 'RGB')
+        orig = orig_img.convert(res_mode)
     elif not orig.mode.startswith('RGB'):
         raise RuntimeError("'trollimage' 1.6+ required to support adding "
                            "overlays/decorations to non-RGB data.")
-    img = orig.pil_image(fill_value=fill_value)
-    cw_ = ContourWriterAGG(coast_dir)
-    cw_.add_coastlines(img, area, outline=color,
-                       resolution=resolution, width=width, level=level_coast)
-    cw_.add_borders(img, area, outline=color,
-                    resolution=resolution, width=width, level=level_borders)
-    # Only add grid if major_lonlat is given.
-    if grid and 'major_lonlat' in grid and grid['major_lonlat']:
-        major_lonlat = grid.pop('major_lonlat')
-        minor_lonlat = grid.pop('minor_lonlat', major_lonlat)
 
-        cw_.add_grid(img, area, major_lonlat, minor_lonlat, **grid)
+    if overlays is None:
+        # fill with sensible defaults
+        general_params = {'outline': color or (0, 0, 0),
+                          'width': width or 0.5}
```

Comment:
Why are these sensible defaults not the defaults in pycoast?

Reply:
Haha good question :) I just took what we had before in this method